File: GREEDY/3_MULPLUS/mulplus.py
data = input()
result = int(data[0])

for i in range(1, len(data)):
    num = int(data[i])
    if num <= 1 or result <= 1:
        result += num
    else:
        result *= num

print(result)

# s = input()
# result = 0

# for char in s:
#     n = int(char)
#     if n <= 1:
#         result += n
#     else:
#         if result <= 1:
#             result += n
#         else:
#             result *= n

# print(result)

File: GREEDY/3_MULPLUS/mulplus.cpp
#include <bits/stdc++.h>

using namespace std;

string str;

int main(void) {
    cin >> str;
    
    long long result = str[0] - '0';

    for (int i=1; i<str.size(); i++) {
        int num = str[i] - '0';
        if (num <= 1 || result <= 1) {
            result += num;
        } else {
            result *= num;
        }
    }
    
    cout << result << endl;
}